src/components/UserSelector.tsx

```diff
@@ -1,16 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onSelectedUser: (val: User | null) => void;
+  selectedUser: User | null;
+  onSelectedPost: (val: Post | null) => void;
```

Comment:
It seems like `onSelectedPost` prop is not used in this component. If it's not used, it's better to remove it from the props to keep the code clean and understandable. Always remember to remove unused code to improve readability and maintainability.